### ISSUE TITLE
API

### ISSUE THREAD

**Issue by ZiGaMi** [GeneralEmbeddedCLibraries/button#1]
Button API shall have:

```C
button_status_t button_init   	(void);
button_status_t button_is_init  (bool * const p_is_init);
button_status_t button_hndl	(void);
button_status_t button_get    	(const button_num_t num,  button_state_t * const p_state);

void button_press_callback(const button_num_t num);
void button_long_press_callback(const button_num_t num);
void button_release_callback(const button_num_t num);
void button_long_release_callback(const button_num_t num);
```

**Comment by ZiGaMi** [comment]
Complete button library shall be configurable via single table.

FILE: button_cfg.h
```C
/**
 *  Button enumerations
 */
typedef enum
{
	eBUTTON_USER = 0,
        eBUTTON_RE,
} button_num_t;

```

FILE: button_cfg.c

```C

// Configuration table ...

```

**Comment by ZiGaMi** [comment]
Solved in release V1.0.0
